include/envoy/network/connection_handler.h

```diff
@@ -97,9 +97,19 @@ class ConnectionHandler {
     virtual Listener* listener() PURE;
 
     /**
-     * Destroy the actual Listener it wraps.
+     * Temporarily stop listening according to implementation's own definition.
      */
-    virtual void destroy() PURE;
+    virtual void pauseListening() PURE;
+
+    /**
+     * Resume listening according to implementation's own definition.
+     */
+    virtual void resumeListening() PURE;
+
+    /**
+     * Stop listening according to implementation's own definition.
+     */
+    virtual void stopListening() PURE;
```

Comment:
`pause` vs. `stop` is kind of confusing. Should the existing `stopListening` become `destroy()`? `shutdown()`? WDYT?

Reply:
Do you mean ConnectionHandler::stopListeners()? I'm not good at naming at all. IMO what the existing stopListeners() really means is stop accepting. The TCP implementation is calling the existing destroy() which makes sense as a TCP listener only listen on new connection. But the UDP implementation stop accepting doesn't mean destroying the listener. That's why I renamed destroy() to stopListening() and let TCP and UDP implementation divert. The existing stopListeners() is an appropriate name for TCP but not for UDP.

I agree that stop and pause aren't informative enough. I named them so to indicate the difference is the former is irreversible and the latter is reversible. This is corresponding to ConnectionHandler's stopListeners() and enable|disableListeners().  Do you have better naming strategy for them?

Reply:
How about `shutdownListening()`? It's minor but I think it makes it more clear. And then int he doc comments maybe more clearly indicate the difference between shutdown/pause? WDYT?